crates/nu-cmd-extra/src/extra/bits/rotate_left.rs

```diff
@@ -96,61 +111,93 @@ impl Command for BitsRol {
                     Span::test_data(),
                 )),
             },
+            Example {
+                description: "rotate left binary data",
+                example: "0x[c0 ff ee] | bits rol 10",
+                result: Some(Value::binary(vec![0xff, 0xbb, 0x03], Span::test_data())),
+            },
         ]
     }
 }
 
-fn get_rotate_left<T: Display + PrimInt>(val: T, bits: u32, span: Span) -> Value
-where
-    i64: std::convert::TryFrom<T>,
-{
-    let rotate_result = i64::try_from(val.rotate_left(bits));
-    match rotate_result {
-        Ok(val) => Value::int(val, span),
-        Err(_) => Value::error(
-            ShellError::GenericError {
-                error: "Rotate left result beyond the range of 64 bit signed number".into(),
-                msg: format!(
-                    "{val} of the specified number of bytes rotate left {bits} bits exceed limit"
-                ),
-                span: Some(span),
-                help: None,
-                inner: vec![],
-            },
-            span,
-        ),
-    }
-}
+fn action(input: &Value, args: &Arguments, span: Span) -> Value {
+    let Arguments {
+        signed,
+        number_size,
+        bits,
+    } = *args;
 
-fn operate(value: Value, bits: usize, head: Span, signed: bool, number_size: NumberBytes) -> Value {
-    let span = value.span();
-    match value {
+    match input {
         Value::Int { val, .. } => {
             use InputNumType::*;
-            // let bits = (((bits % 64) + 64) % 64) as u32;
-            let bits = bits as u32;
-            let input_type = get_input_num_type(val, signed, number_size);
-            match input_type {
-                One => get_rotate_left(val as u8, bits, span),
-                Two => get_rotate_left(val as u16, bits, span),
-                Four => get_rotate_left(val as u32, bits, span),
-                Eight => get_rotate_left(val as u64, bits, span),
-                SignedOne => get_rotate_left(val as i8, bits, span),
-                SignedTwo => get_rotate_left(val as i16, bits, span),
-                SignedFour => get_rotate_left(val as i32, bits, span),
-                SignedEight => get_rotate_left(val, bits, span),
+            let val = *val;
+            let bits = bits as u64;
```

Comment:
Since `bits` is always converted to a `u32`, perhaps it could be converted to a `u32` right here? (and an error thrown if it's bigger than a `u32`?)